Sources/MongoSwift/BSON/BSONDocument.swift

```diff
@@ -145,10 +145,10 @@ extension BSONDocument {
      * presence first.
      *
      * - Throws:
-     *   - `MongoError.InternalError` if the new value is an `Int` and cannot be written to BSON.
-     *   - `MongoError.LogicError` if the new value is a `BSONDecimal128` or `BSONObjectID` and is improperly formatted.
-     *   - `MongoError.LogicError` if the new value is an `Array` and it contains a non-`BSONValue` element.
-     *   - `MongoError.InternalError` if the underlying `bson_t` would exceed the maximum size by encoding this
+     *   - `BSONError.InternalError` if the new value is an `Int` and cannot be written to BSON.
```

Comment:
this seems like an error we don't need to worry about anymore... I think it's probably a remnant of past API when `Int` was a `BSONValue`